subnet_sharing.tf

```diff
@@ -5,5 +5,5 @@ module "subnet_sharing" {
   prepend_resource_type = var.prepend_resource_type
   private_subnet_arns   = var.share_private_subnets ? toset(aws_subnet.private[*].arn) : []
   public_subnet_arns    = var.share_public_subnets ? toset(aws_subnet.public[*].arn) : []
-  tags                  = var.subnet_sharing_custom_tags
+  tags                  = merge(var.tags, var.subnet_sharing_custom_tags)
```

Comment:
I think that the merge should be done on the var.subnet_sharing_custom_tags level and not here as this level is hidden and would just enforce the VPC tags on the subnet sharing even when that makes no sense.